framework/include/outputs/ConsoleUtils.h

```diff
@@ -57,7 +57,7 @@ std::string outputAuxiliarySystemInformation(FEProblemBase & problem);
 /**
  * Output the Nonlinear system information
  */
-std::string outputNonlinearSystemInformation(FEProblemBase & problem);
+std::string outputNonlinearSystemInformation(FEProblemBase & problem, unsigned int nl_sys_num);
```

Comment:
Pedantically someone looking at this API to determine whether they want to call it or not will not care whether the `unsigned int` argument is `const` or not because it is passed-by-value ... the caller does not have to worry about whether the method modifies the incoming argument because it is passed-by-value. And then in the implementation, if you actually don't modify the argument, you can mark the argument as `const` if you want since it does not alter the signature of the method. So if you look at the `.C` implementation you will likely see that I've done that

Reply:
But I do see some value in conveying the implementation in the header